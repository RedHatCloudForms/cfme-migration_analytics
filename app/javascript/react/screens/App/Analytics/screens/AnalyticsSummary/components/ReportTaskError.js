import React from 'react';
import PropTypes from 'prop-types';
import EmptyStateWithButton from './EmptyStateWithButton';

const ReportTaskError = ({ taskWithError, onRetryClick }) => (
  <EmptyStateWithButton
    iconName="error-circle-o"
    title={__('Failed to run summary report')}
    message={
      <React.Fragment>
        {__('Task failed: ')}
        {taskWithError.name}
        <br />
        {__('Error message: ')}
        {taskWithError.message}
      </React.Fragment>
    }
    buttonText={__('Try again')}
    onClick={onRetryClick}
  />
);

ReportTaskError.propTypes = {
  taskWithError: PropTypes.shape({
    name: PropTypes.string,
    message: PropTypes.string
  }).isRequired,
  onRetryClick: PropTypes.func.isRequired
};

export default ReportTaskError;
