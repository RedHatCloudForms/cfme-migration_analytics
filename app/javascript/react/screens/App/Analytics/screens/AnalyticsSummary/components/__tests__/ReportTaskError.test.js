import React from 'react';
import { shallow } from 'enzyme';
import ReportTaskError from '../ReportTaskError';

describe('report task error', () => {
  const getBaseProps = () => ({
    taskWithError: {
      name: 'task name here',
      message: 'some error message'
    },
    onRetryClick: jest.fn()
  });

  test('renders correctly', () => {
    const component = shallow(<ReportTaskError {...getBaseProps()} />);
    expect(component).toMatchSnapshot();
  });
});
