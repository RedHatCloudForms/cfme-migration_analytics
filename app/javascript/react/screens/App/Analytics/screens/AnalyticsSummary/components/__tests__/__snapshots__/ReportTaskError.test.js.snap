// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`report task error renders correctly 1`] = `
<EmptyStateWithButton
  buttonText="Try again"
  href={null}
  iconName="error-circle-o"
  iconType="pf"
  message={
    <React.Fragment>
      Task failed: 
      task name here
      <br />
      Error message: 
      some error message
    </React.Fragment>
  }
  onClick={[MockFunction]}
  title="Failed to run summary report"
/>
`;
